### PR TITLE
client: Add MOVE and UID MOVE support

### PR DESCRIPTION
**Pull request by mtorromeo**
This implements the MOVE and UID MOVE commands as defined in the "Internet Message Access Protocol (IMAP) - MOVE Extension" [RFC6851](https://tools.ietf.org/html/rfc6851#section-3).

### CODE REVIEW

**Comment by mtorromeo** [comment]
Note that I couldn't use `move` as the name of the function due to it being a reserved keyword in rust. I opted for `imap_move` but I can change it to something better if you want.

File: src/client.rs [hunk 1]
```diff
@@ -1173,6 +1187,40 @@ mod tests {
...
+            * 2 EXPUNGE\r\n\
+            * 1 EXPUNGE\r\n\
+            a1 OK Move completed\r\n".to_vec();
+        let mailbox_name = "MEETING";
+        let command = format!("a1 MOVE 1:2 {}\r\n", quote!(mailbox_name));
+        let mock_stream = MockStream::new(response);
+        let mut session = mock_session!(mock_stream);
+        session.imap_move("1:2", mailbox_name).unwrap();
+        assert!(
+            session.stream.get_ref().written_buf == command.as_bytes().to_vec(),
```
Comment (jonhoo) [inline src/client.rs:1207]: Should probably be `assert_eq!`

File: src/client.rs [hunk 2]
```diff
@@ -1173,6 +1187,40 @@ mod tests {
...
+        let response = b"* OK [COPYUID 1511554416 142,399 41:42] Moved UIDs.\r\n\
+            * 2 EXPUNGE\r\n\
+            * 1 EXPUNGE\r\n\
+            a1 OK Move completed\r\n".to_vec();
+        let mailbox_name = "MEETING";
+        let command = format!("a1 UID MOVE 41:42 {}\r\n", quote!(mailbox_name));
+        let mock_stream = MockStream::new(response);
+        let mut session = mock_session!(mock_stream);
+        session.uid_move("41:42", mailbox_name).unwrap();
+        assert!(
```
Comment (jonhoo) [inline src/client.rs:1223]: Should probably also be `assert_eq!`

File: src/client.rs [hunk 3]
```diff
@@ -563,6 +563,20 @@ impl <T: Read + Write> Session<T> {
         self.run_command_and_check_ok(&format!("UID COPY {} {}", uid_set, mailbox_name))
     }
 
+    /// Moves each message in the sequence into the destination mailbox.
+    pub fn imap_move(&mut self, sequence_set: &str, mailbox_name: &str) -> Result<()> {
```
Comment (jonhoo) [inline src/client.rs]: `move` being reserved is unfortunate here, it's true... Perhaps `mv`?

**Comment by jonhoo** [comment]
I wonder if the docstring for the two commands should also include a link to the relevant part of the RFC?

**Comment by mtorromeo** [comment]
If we add rfc links to this docstrings, should we add it to all the other functions for consistency?

File: src/client.rs (hunk 3 above)
Comment (mtorromeo) [reply src/client.rs]: I don't have a strong preference either way. It will still remain the only function that does not match the IMAP command sadly...

**Comment by jonhoo** [comment]
I think (but could be wrong) all your PRs have been for features that are IMAP _extensions_. That's why I wanted to call out the reference specifically. I think it'd be great to add IMAP RFC links to all the methods (see also #62), though that's outside the scope of this PR :)

**Comment by mtorromeo** [comment]
> I think (but could be wrong) all your PRs have been for features that are IMAP _extensions_

Not the SEARCH/UID SEARCH commands.

I'll add the references in the docstrings for the extensions then.

File: src/client.rs (hunk 1 above)
Comment (mtorromeo) [reply src/client.rs:1207]: See #87

File: src/client.rs (hunk 2 above)
Comment (mtorromeo) [reply src/client.rs:1223]: See #87

File: src/client.rs (hunk 3 above)
Comment (mtorromeo) [reply src/client.rs]: Renamed to mv

**Comment by mtorromeo** [comment]
RFC links added

**Comment by jonhoo** [comment]
Let's make `uid_move` be `uid_mv` now too for consistency. And then I'm happy to merge!

**Comment by jonhoo** [comment]
Awesome, thank you!

**Pull request merged**